hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -81,11 +81,16 @@ def check_sample(
     return tuple(values)
 
 
+# TODO probably move this to a method on ConjectureData
```

Comment:
Yeah, functions which accept a mandatory first argument which is an instance of some class... are just messy methods.  Putting them all on the class might reveal that the class is too big, but is IMO still an improvement.

(historically, this probably dates back to when David was first learning Python + writing Hypothesis 0.x)